presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveBucketedTables.java

```diff
@@ -93,6 +123,154 @@ public void testAllowMultipleFilesPerBucket()
+    {
+        String tableName = mutableTableInstanceOf(BUCKETED_NATION).getNameInDatabase();
+        assertThat(query(format("SELECT count(*) FROM %s", tableName)))
+                .containsExactly(row(0));
+    }
+
+    @Test
+    public void testSelectFromIncompleteBucketedTableEmptyTablesAllowed()
+    {
+        String tableName = mutableTableInstanceOf(BUCKETED_NATION).getNameInDatabase();
+        populateRowToHiveTable(tableName, ImmutableList.of("2", "'name'", "2", "'comment'"), Optional.empty());
+        // insert one row into nation
+        assertThat(query(format("SELECT count(*) from %s", tableName)))
+                .containsExactly(row(1));
+        assertThat(query(format("select n_nationkey from %s where n_regionkey = 2", tableName)))
+                .containsExactly(row(2));
+    }
+
+    @Test
+    public void testInsertPartitionedBucketed()
+    {
+        String tableName = mutableTablesState().get(BUCKETED_NATION_PREPATED).getNameInDatabase();
+
+        String ctasQuery = "CREATE TABLE %s WITH (bucket_count = 4, bucketed_by = ARRAY['n_regionkey'], partitioned_by = ARRAY['part_key']) " +
+                "AS SELECT n_nationkey, n_name, n_regionkey, n_comment, n_name as part_key FROM %s";
+        query(format(ctasQuery, tableName, NATION.getName()));
+
+        assertThat(query(format("SELECT count(*) FROM %s", tableName))).containsExactly(row(25));
+        assertThat(query(format("SELECT count(*) FROM %s WHERE n_regionkey=0", tableName))).containsExactly(row(5));
+        assertThat(query(format("SELECT count(*) FROM %s WHERE part_key='ALGERIA'", tableName))).containsExactly(row(1));
+        assertThat(query(format("SELECT count(*) FROM %s WHERE n_regionkey=0 AND part_key='ALGERIA'", tableName))).containsExactly(row(1));
+    }
+
+    @Test
+    public void testCreatePartitionedBucketedTableAsSelect()
+    {
+        String tableName = mutableTablesState().get(BUCKETED_PARTITIONED_NATION).getNameInDatabase();
+
+        query(format("INSERT INTO %s SELECT n_nationkey, n_name, n_regionkey, n_comment, n_name FROM %s", tableName, NATION.getName()));
+
+        assertThat(query(format("SELECT count(*) FROM %s", tableName))).containsExactly(row(25));
+        assertThat(query(format("SELECT count(*) FROM %s WHERE n_regionkey=0", tableName))).containsExactly(row(5));
+        assertThat(query(format("SELECT count(*) FROM %s WHERE part_key='ALGERIA'", tableName))).containsExactly(row(1));
+        assertThat(query(format("SELECT count(*) FROM %s WHERE n_regionkey=0 AND part_key='ALGERIA'", tableName))).containsExactly(row(1));
+    }
+
+    @Test
+    public void testInsertIntoBucketedTables()
+    {
+        String tableName = mutableTablesState().get(BUCKETED_NATION).getNameInDatabase();
+
+        assertThat(() -> query(format("INSERT INTO %s SELECT * FROM %s", tableName, NATION.getName())))
+                .failsWithMessage("Cannot insert into bucketed unpartitioned Hive table");
```

Comment:
Is it true now that https://github.com/prestosql/presto/pull/924 is merged?
Please rebase to force Travis re-merge anew.